### PR TITLE
Browser Passthrough - Improving Yogs user experience by 10x

### PR DESCRIPTION
**Pull request by monster860**
Being focused on a browser window is such a disruptive experience. If you want to go back to doing spaceman-like things like swapping hands, moving around, etc, you have to click on the BYOND window and unfocus the browser before you can actually do what you want with your hotkeys - a significant disruption.

Imagine the following: You're hacking some door, so you pull out a screwdriver in one hand, multitool in the other. You open the panel,  select the multitool, and hit the "pulse" button for the wire you want, but when you hit X to swap hands to go back to the screwdriver to close the panel, nothing happens! Maybe it's because you focused on the browser window? You have to click on the game view. This PR alleviates that.

Here's an even simpler scenario: You're in the middle of hacking a door, when security shows up. Now you gotta run away! But oh no, the WASD/arrow keys don't do anything because you're focused on the wires window! There are many such interactions that make people think "wow, this is such a clunky user interface".

Here's a bonus video of a nuke op purchasing a weapon while being chased by an imaginary security officer: https://streamable.com/05ycm

:cl: monster860
add: You can now use hotkeys while focused on browser windows
/:cl:

### CODE REVIEW

File: tgui/assets/tgui.js [hunk 1]
```diff
@@ -6,16 +6,16 @@ try{!a&&s["return"]&&s["return"]()}finally{if(r)throw i}}return n},_=function(t)
...
 name:"css",extend:function(t,e,n){if(n.css){var a=qp++,r=n.noCssTransform?n.css:jp(n.css,a);e.cssId=a,Op.add({id:a,styles:r})}},init:function(){}},Vp=Gp,zp={name:"data",extend:function(t,e,n){var a=void 0,r=void 0;if(n.data&&u(n.data))for(a in n.data)r=n.data[a],r&&"object"==typeof r&&(u(r)||i(r))&&m("Passing a `data` option with object and array properties to Ractive.extend() is discouraged, as mutating them is likely to cause bugs. Consider using a data function instead:\n\n  // this...\n  data: function () {\n    return {\n      myObject: {}\n    };\n  })\n\n  // instead of this:\n  data: {\n    myObject: {}\n  }");e.data=Bt(e.data,n.data)},init:function(t,e,n){var a=Bt(t.prototype.data,n.data);return"function"==typeof a&&(a=a.call(e)),a||{}},reset:function(t){var e=this.init(t.constructor,t,t.viewmodel);return t.viewmodel.reset(e),!0}},Wp=zp,Hp=null,Kp=["preserveWhitespace","sanitize","stripComments","delimiters","tripleDelimiters","interpolate"],Qp={fromId:zt,isHashedId:Wt,isParsed:Ht,getParseOptions:Kt,createHelper:Gt,parse:Vt},Yp=Qp,$p={name:"template",extend:function(t,e,n){var a;"template"in n&&(a=n.template,"function"==typeof a?e.template=a:e.template=Jt(a,e))},i
```
Comment (ghost) [inline tgui/assets/tgui.js:13]: I'm not a reviewer, I just have a question:

Is this just a tgui rebuild? Or is something other than the code in this PR being added?

Comment (monster860) [reply tgui/assets/tgui.js:13]: just a tgui rebuild

**Review by Amelia0010 — APPROVED** [review]

File: tgui/assets/tgui.js (hunk 1 above)
Comment (ghost) [reply tgui/assets/tgui.js:13]: thanks, now I can yoink this for hippie

**Review by alexkar598 — DISMISSED** [review]
Pressing a key, clicking on a text box while said key is pressed will put the key in a locked down state until pressed again, please register the keyUp event in text boxes.

**Comment by ghost** [comment]
@alexkar598 99% sure monster said something about this PR not doing anything about text boxes (they're not actually browser windows per se, I don't think). I may have misunderstood him however, it'd be better to get a reply from him first.

**Comment by ynot01** [comment]
its a tgui thing i dont think non-tgui is affected

**Comment by alexkar598** [comment]
@HideAndSeekLOGIC 99% sure I CAN READ FUCKING CODE AND IT DOESNT TRIGGER THE EVENTS IF ITS A TEXT BOX!

-Love, alexar598

**Comment by monster860** [comment]
When it comes to text boxes in a UI window, if I hold down a key, click on the text box and let go I'll stop moving when I let go because (probably) the keyup event happens on the same target as the keydown.

of course if I click on an `input()` box, that's not a browser window, that's a special snowflake BYOND thingy that doesn't have it.

**Comment by ghost** [comment]
@alexkar598 shut

**Comment by Hopekz** [comment]
I just played around with this on a test server. 
Really good PR +1 .
**Is there any way to make this work on the PDA interface screen?**

**Comment by LoliconSlayer** [comment]
If this doesn't get speedmerged right fucking now I'm gonna riot

**Comment by Nickvr628** [comment]
what about non-hotkey mode users

**Comment by monster860** [comment]
I added this functionality to the PDA.

**Comment by MegaEmpirical** [comment]
Fucking yes.

**Pull request merged**